### ISSUE TITLE
ios_config: Error if `#macro` is specified

### ISSUE THREAD

**Issue by gundalow** [ansible/ansible#38591]
##### ISSUE TYPE
 - Bug Report
 - Documentation Report

##### COMPONENT NAME
ios_config

##### ANSIBLE VERSION
```
2.5.0
```

##### SUMMARY

Following on from https://github.com/ansible/ansible/issues/38420

`ios_config` should detect if one of the `lines:` starts with `#macro` and fail the task with a message stating the macros are not supported. The work around is detailed in https://github.com/ansible/ansible/issues/38420 which should be part of message.

Should also update module docs

##### STEPS TO REPRODUCE
```yaml
---

- hosts: access_switches

  tasks:

  - name: MACRO PHONE
    ios_config:
      lines:
      - 'switchport voice vlan 60'
      - 'switchport port-security maximum 2'
      - 'spanning-tree portfast'
      - 'no cdp enable'
      - 'no macro desc'
      - '#macro keywords $DESCRIPTION'
      - '@'
      parents: macro name PHONE
      before: no macro name PHONE
      after: exit
      match: line
      replace: block

```

##### EXPECTED RESULTS

Meaningful message stating this isn't supported

##### ACTUAL RESULTS
```

```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/network/ios/ios_config.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/ios/ios_config.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @privateip @rcarrillocruz
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by abenokraitis** [comment]
@gundalow is this still needed and high priority?

**Comment by ansibot** [comment]
@gundalow: Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information.

Here are the items we could not find in your description:
- issue type
- ansible version
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @justjais
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@gundalow This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@gundalow This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@gundalow You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
